core/src/main/java/org/elasticsearch/plugins/ScriptPlugin.java

```diff
@@ -39,6 +40,15 @@ default ScriptEngineService getScriptEngineService(Settings settings) {
     }
 
     /**
+     * Returns a {@link TemplateService.Backend} if this plugin implements a template backend or null if it doesn't. Note that Elasticsearch
+     * will refuse to start if there is more than one template backend and it is bundled with Mustache. To replace that backend you'd have
+     * to remove the Mustache module which is super untested.
```

Comment:
That was pretty much my intent. Templates are in a module for permission isolation reasons but, yeah, we really don't support anything else. You could try but it isn't likely to work.

Would it be good enough to make this comment reflect that better?